src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ForegroundDispatcher.cs

```diff
@@ -20,13 +26,24 @@ public virtual void AssertForegroundThread([CallerMemberName] string caller = nu
             if (!IsForegroundThread)
```

Comment:
I eventually want to rename `IsForegroundThread` to `IsSpecializedForegroundThread` and `AssertForegroundThread` to `AssertSpecializedForegroundThread` to clearly indicate that the referred-to foreground thread is distinct from the UI thread. However, I tried doing this and the diff was huge, so I plan to do it in a follow-up PR.

Reply:
Will do in a follow-up!